gorm-rest/src/main/groovy/yakworks/rest/gorm/controller/EntityResponder.groovy

```diff
@@ -110,14 +112,22 @@ class EntityResponder<D> {
         // Map pclone = Maps.clone(parms) as Map<String, Object>
         //remove the fields that grails adds for controller and action
         // pclone.removeAll {it.key in whitelistKeys }
-
-        QueryArgs qargs = QueryArgs.of(pager)
-        //require q if its set
-        if(pathItem?.qRequired) qargs.qRequired = true
-        qargs = qargs.build(parms)
-        qargs.validateQ()
-        if(debugEnabled) log.debug("QUERY ${entityClass.name} queryArgs.criteria: ${qargs.criteria}")
-        ((QueryMangoEntityApi)getRepo()).queryList(qargs, null, debugEnabled ? log : null)
+        try {
+            QueryArgs qargs = QueryArgs.of(pager)
+            //require q if its set
+            if (pathItem?.qRequired) qargs.qRequired = true
+            qargs = qargs.build(parms)
+            qargs.validateQ()
+            if (debugEnabled) log.debug("QUERY ${entityClass.name} queryArgs.criteria: ${qargs.criteria}")
+            ((QueryMangoEntityApi) getRepo()).queryList(qargs, null, debugEnabled ? log : null)
+        } catch (JsonException | IllegalArgumentException | QueryException ex) {
+            //See #1925 - Catch bad query in 'q' parameter and report back. So we dont pollute logs, and can differentiate that its not us.
+            //Hibernate throws IllegalArgumentException when Antlr fails to parse query
+            //and throws QueryException when hibernate fails to execute query
+            throw DataProblem.ex("Invalid query $ex.message")
+        } catch (DataAccessException ex) {
+            throw DataProblem.of(ex).toException()
```

Comment:
this seems like its not needed. the ex is is already just throwable. dont need to create a new one